### PR TITLE
added functionality for dynamic sidenav used on documentation section…

### PR DESCRIPTION
**Pull request by jamesthebrooks**
Added dynamic sidenav used for documents section as well as contact us page

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -194,6 +194,102 @@ module.exports = function (config) {
...
+   *   In your file front matter, to opt in to be included in a menu:
+   *   sidenav: true,
+   *   sidenavIdentifier: "docs",
+   */
+
+  config.addCollection("sideNavigation", function (collectionApi) {
+    const basePath = "content";
+    const mainFolders = fs
+      .readdirSync(basePath, {withFileTypes: true})
+      .filter((dirent) => {
```
Comment (sknep) [inline .eleventy.js:217]: is `dirent` a typo of `direct` ?

Comment (jamesthebrooks) [reply .eleventy.js:217]: It's shorthand for "DIRectory ENTry" and probably comes from old C, but the JavaScript Type for it is also capitalized "Dirent".  Can change the variable to whatever though as it's used arbitrarily here.  I don't have much of an opinion on it.

Comment (sknep) [reply .eleventy.js:217]: Interesting! I don't haver a classical C background so I didn't recognize it. Fine with me to keep!

File: _includes/sidenav.html [hunk 2]
```diff
@@ -1,51 +1,39 @@
...
-          </li>
-          <li class="usa-sidenav__item">
-            <a href="">Child link</a>
-          </li>
+    {% assign categories = collections.sideNavigation | getSideNav: sidenavIdentifier %}
+    <nav>
+        <ul class="usa-sidenav">
+            {%- for category in categories -%}
+            <li class="usa-sidenav__item">
+                <a class="usa-sidenav__item--parent-item usa-sidenav__sublist--middle-generation {% if category.id == parent %}usa-current{% endif %}"
```
Comment (sknep) [inline _includes/sidenav.html:12]: I think you just noticed this but we lost the `href` on this a tag

File: .eleventy.js [hunk 3]
```diff
@@ -194,6 +194,102 @@ module.exports = function (config) {
...
+            });
+          }
+        }
+        result.push({
+          ...collection,
+          categories,
+        });
+      }
+    }
+    console.log('Dynamic Side Navigation Configurations', result);
```
Comment (sknep) [inline .eleventy.js:289]: do we need to keep the .log for every build or was that for debugging?

File: js/app.js [hunk 4]
```diff
@@ -1,74 +1,101 @@
...
-const subLists = document.querySelectorAll("ul.usa-sidenav__sublist");
-const middleSubList = document.querySelectorAll(".usa-sidenav__sublist ul");
 
-if (!!anchors) {
+function ready(callback) {
+  if (document.readyState !== 'loading') { // in case the document is already rendered
+    callback();
+  } else if (document.addEventListener) { // modern browsers
+    document.addEventListener('DOMContentLoaded', callback);
+  } else { // IE <= 8
```
Comment (sknep) [inline js/app.js:11]: its ok to leave this in, but we don't have to support anything older than IE 11 going forward

**Review by sknep — APPROVED** [review]
Approved -- but do let me know what you think about the two questions

**Pull request merged**